docs/developer-guide/etendo-classic/concepts/Processes.md

```diff
@@ -0,0 +1,292 @@
+
+!!!info
+    For more information, read [How to create a Background Process](../../../developer-guide/etendo-classic/how-to-guides/How_to_create_a_Background_Process.md).
+
+In this window you can define a background process. The process to be executed can be selected in the *Process* field. Then you define the scheduling for the background process and it is then ready to use.
+
+!!!note
+    There is no user interaction necessary to execute the process, thus, no pop-up will appear prompting for additional parameters.
+
+#####  Kill a background Process
+ 
+  
+A button *Kill Process* is shown in the Process Monitor window on that processes that implement the KillableProcess interface while a process is being executed (Status = Processing). The interface KillableProcess will let you kill your process using a kill mechanism.
+
+**Mechanism to Kill**
+
+The mechanism that your background process uses to kill itself might vary between implementations. However, the main idea in any implementation should be to check some flag perodically during execution to see if a kill has been requested, and if the flag is set, somehow abort the execution of the rest of the job's work.
+
+
+Let's see an example,
+
+here we have a dummy process which simply prints the identifier in the log for all the business partners:
+
+    
+    
+    public class DummyProcess extends DalBaseProcess {
+     
+      private static final Logger log4j = Logger.getLogger(DummyProcess.class);
+     
+      @Override
+      protected void doExecute(ProcessBundle bundle) throws Exception {
+        try {
+     
+          // Get all business partners
+          final OBCriteria<BusinessPartner> bpCri = OBDal.getInstance().createCriteria(
+              BusinessPartner.class);
+          bpCri.setFetchSize(1000);
+          final ScrollableResults partnerScroller = bpCri.scroll(ScrollMode.FORWARD_ONLY);
+          int i = 1;
+          // Loop all business partners using a ScrollabeResults to avoid performance issues
+          while (partnerScroller.next()) {
+            final BusinessPartner bp = (BusinessPartner) partnerScroller.get()[0];
+            // Print the Identifier for every business partner
+            log4j.info(bp.getIdentifier());
+     
+            if ((i % 100) == 0) {
+              OBDal.getInstance().getSession().clear();
+            }
+            i++;
+     
+          }
+          partnerScroller.close();
+        } catch (Exception ex) {
+          Throwable e = DbUtility.getUnderlyingSQLException(ex);
+          log4j.error("Error in DummyProcess", e);
+          throw new JobExecutionException(e.getMessage(), e);
+        }
+     
+      }
+     
+    }
+
+**Killable Process**
+
+Now, we will see the same process but with Interface KillableProcess implemented with the kill method.
+
+    
+    
+    public class DummyProcessKillable extends DalBaseProcess implements KillableProcess {
```

Comment:
Dejo el código completo que saque del módulo  (tiene los imports )

``` java title="DummyProcessKillable.java"
package com.openbravo.example.killprocess.process;

import org.apache.log4j.Logger;
import org.hibernate.ScrollMode;
import org.hibernate.ScrollableResults;
import org.openbravo.dal.service.OBCriteria;
import org.openbravo.dal.service.OBDal;
import org.openbravo.model.common.businesspartner.BusinessPartner;
import org.openbravo.scheduling.KillableProcess;
import org.openbravo.scheduling.ProcessBundle;
import org.openbravo.service.db.DalBaseProcess;
import org.openbravo.service.db.DbUtility;
import org.quartz.JobExecutionException;

public class DummyProcessKillable extends DalBaseProcess implements KillableProcess {

  private static final Logger log4j = Logger.getLogger(DummyProcessKillable.class);

  // Add a variable 'stop' to control the kill implementation and set false by default
  private boolean stop = false;

  @Override
  protected void doExecute(ProcessBundle bundle) throws Exception {
    try {
      Thread.sleep(30000);
      // Get all business partners
      final OBCriteria<BusinessPartner> bpCri = OBDal.getInstance().createCriteria(
          BusinessPartner.class);
      bpCri.setFetchSize(1000);
      final ScrollableResults partnerScroller = bpCri.scroll(ScrollMode.FORWARD_ONLY);
      int i = 1;
      // Loop all business partners using a ScrollabeResults to avoid performance issues

      // Only continue with the process if the variable 'stop' is false
      while (partnerScroller.next() && !stop) {
        final BusinessPartner bp = (BusinessPartner) partnerScroller.get()[0];
        // Print the Identifier for every business partner
        log4j.info(bp.getIdentifier());
        // Add a timeout of 30 seconds
        Thread.sleep(30000);

        if ((i % 100) == 0) {
          OBDal.getInstance().getSession().clear();
        }
        i++;

      }
      partnerScroller.close();
    } catch (Exception ex) {
      Throwable e = DbUtility.getUnderlyingSQLException(ex);
      log4j.error("Error in DummyProcess", e);
      throw new JobExecutionException(e.getMessage(), e);
    }

  }

  @Override
  public void kill() throws Exception {
    log4j.info("Kill Dummy Process Killable");
    // When kill is called set variable 'stop' to true so the process will be interrupted in the
    // next iteration: while (partnerScroller.next() && !stop)
    stop = true;
  }

}

```